### PR TITLE
Get the benchmarks running again

### PR DESCRIPTION
**Pull request by stevenaw**
Fixes https://github.com/nunit/nunit/issues/4806

This PR gets the benchmarks running again by following the steps outlined in the issue. I was able to inject the assemblyconfiguration at runtime by defining a BenchmarkDotNet Job based on the default one

### CODE REVIEW

**Comment by stevenaw** [comment]
This is ready for review. 

However, this could be taken one step further by updating the TFM from `net6.0` to something newer. Should it be updated to latest (NET 9.0) or latest LTS (net8.0)? Note that latest LTS will hit EOL later I believe

File: src/NUnitFramework/benchmarks/nunit.framework.benchmarks/nunit.framework.benchmarks.csproj
Comment (manfred-brands) [inline src/NUnitFramework/benchmarks/nunit.framework.benchmarks/nunit.framework.benchmarks.csproj:1]: Suggest to change target to net8.0 as being the currently support LTS and target framework

**Review by manfred-brands — CHANGES_REQUESTED** [review]
I currently get compile time errors when DotNetBenchMark is compiling the code, i.e. when running.
MSBuild wants `GenerateDocumentationFile` be set. We have it set on all other projects, I think we should move it to `Directory.Build.props` instead. For some reason MS can only check for 'unused namespace' if documentation is enabled.
However setting that results in CS1591 errors as there is no documentation, we have disabled that in various other test projects.

**Comment by stevenaw** [comment]
Thanks @manfred-brands .
Odd, I didn't get those on the current TFM, but I _had_ when moving to net8.0 initially.

I can make those updates tomorrow. It'll likely also require a `NoWarn` to suppress the warning for missing xmldocs on public members (1591 ?) similar to how our other test projects are already setup.

**Comment by manfred-brands** [comment]
I made them here and can push them if you are ok with that.

**Comment by stevenaw** [comment]
Sure! Feel free, thanks @manfred-brands

**Review by manfred-brands — APPROVED** [review]
I'm happy with this now. Verify it runs for you.

File: src/NUnitFramework/benchmarks/nunit.framework.benchmarks/nunit.framework.benchmarks.csproj [hunk 1]
```diff
@@ -2,14 +2,21 @@
...
         <OutputType>Exe</OutputType>
-        <TargetFrameworks>net6.0</TargetFrameworks>
+        <TargetFrameworks>net8.0</TargetFrameworks>
         <ImplicitUsings>enable</ImplicitUsings>
         <LangVersion>latest</LangVersion>
         <Optimize>true</Optimize>
-        <AssemblyOriginatorKeyFile>..\..\..\nunit.snk</AssemblyOriginatorKeyFile>
         <RootNamespace>NUnit.Framework</RootNamespace>
+        <GenerateDocumentationFile>true</GenerateDocumentationFile>
+        <NoWarn>1591</NoWarn>
```
Comment (mikkelbu) [inline src/NUnitFramework/benchmarks/nunit.framework.benchmarks/nunit.framework.benchmarks.csproj:11]: Is this needed when we also have it in the Directory.Build.props ?

**Comment by stevenaw** [comment]
Thanks @manfred-brands @mikkelbu 
I should be able to get to both of those in a few hours

File: src/NUnitFramework/benchmarks/nunit.framework.benchmarks/nunit.framework.benchmarks.csproj (hunk 1 above)
Comment (manfred-brands) [reply src/NUnitFramework/benchmarks/nunit.framework.benchmarks/nunit.framework.benchmarks.csproj:11]: No, I must have left that in accidentally.

**Comment by stevenaw** [comment]
The requested changes are done and I've confirmed that @manfred-brands 's changes also work for me. 

I'll wait a few days before merging in case anyone wants to take another look.

**Review by mikkelbu — APPROVED** [review]
Still looks good to me :). Thanks for the work @stevenaw and @manfred-brands 👍

**Comment by stevenaw** [comment]
Thanks @mikkelbu !

**Pull request merged**